cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,298 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+DOCUMENTATION = '''
+---
+module: ec2_vpc_nat_gateway
+short_description: Manage AWS VPC NAT Gateways
+description:
+  - Ensure the state of AWS VPC NAT Gateways based on their id, allocation and subnet ids.
+version_added: "2.1"
+options:
+  state:
+    description:
+      - Ensure NAT Gateway is present or absent
+    required: false
+    default: "present"
+    choices: ["present", "absent"]
+  nat_gateway_id:
+    description:
+      - The id AWS dynamically allocates to the NAT Gateway on creation
+    required: false
+    default: None
+  subnet_id:
+    description:
+      - The id of the subnet to create the NAT Gateway in
+    required: false
+    default: None
+  allocation_id:
+    description:
+      - The id of the elastic IP allocation
+    required: false
+    default: None
+  wait:
+    description:
+      - Wait for operation to complete before returning
+    required: false
+    default: true
+  wait_timeout:
+    description:
+      - How many seconds to wait for an operation to complete before timing out
+    required: false
+    default: 300
+
+author:
+  - "Jon Hadfield (@jonhadfield)"
+  - "Karen Cheng(@Etherdaemon)"
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Ensure that a VPC NAT Gateway exists in a subnet by passing the subnet id and EIP allocation id
+ec2_vpc_nat_gateway:
+  state: present
+  subnet_id: subnet-a67643d1
+  allocation_id: eipalloc-a32baec6
+
+# Ensure that a VPC NAT Gateway exists in a subnet by passing the subnet id and
+# EIP allocation id but do not wait for operation to complete before continuing
+ec2_vpc_nat_gateway:
+  state: present
+  subnet_id: subnet-a67643d1
+  allocation_id: eipalloc-a32baec6
+  wait: false
+
+# Ensure that a VPC NAT Gateway identified by id does not exist
+ec2_vpc_nat_gateway:
+  state: absent
+  nat_gateway_id: nat-0d1e3a878585988f8
+'''
+
+RETURN = '''
+nat_gateway_id:
+    description: id of the VPC NAT Gateway
+    returned: In all cases except for when state=absent and it does not exist
+    type: string
+    sample: "nat-0d1e3a878585988f8"
+'''
+
+import time
+import datetime
+try:
+    import boto3
+    import botocore
+    HAS_BOTO_3 = True
+except ImportError:
+    HAS_BOTO_3 = False
+from distutils.version import LooseVersion
+if LooseVersion(botocore.__version__) < LooseVersion("1.3.14"):
+    HAS_SUFFICIENT_BOTOCORE = False
```

Comment:
As per guidelines...

> Use feature testing (e.g. hasattr('boto.module', 'shiny_new_method')) to check whether boto supports a feature rather than version checking

>